src/api/db/project_db.py

```diff
@@ -130,18 +132,26 @@ def mark_project_deleted(project_id: str) -> bool:
     move(filename, new_filename)
     return True
 
-def add_topic_to_project(project_id: str, topic_id: str) -> Optional[Project]:
-    """Add a topic to a project's topic list if the topic exists."""
+def add_topic_to_project(project_id: str, topic_id: str):
+    """
+    Associate a topic with a project.
+    Adjust this logic based on your actual project model and DB functions.
+    """
+    from api.db.project_db import get_project, update_project  
+
     project = get_project(project_id)
     if not project:
-        return None
-            
+        raise HTTPException(status_code=404, detail="Project not found")
```

Comment:
```suggestion
        None
```

The db class might be used outside of the API context, so I would not raise HTTP errors here, but only in the routes. This is what happens in the existing use of this method.